source/Cosmos.Core/AdvancedObject.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cosmos.Core;
+using IL2CPU.API;
+
+namespace Cosmos.Core
+{
+    /// <summary>
+    /// This Class is created for Add more methods to objects by add this to base class
+    /// </summary>
+    public class AdvancedObject
+    {
+        public unsafe uint GetTypeId()
```

Comment:
What do you need this function for? What advantage does it have over just calling `GCImplementation.GetType(object)`? 

More generally could you explain what the use case for AdvancedObject is?

Reply:
I wrote a description on the class file

Reply:
I am still not convinced that we actually need this method.